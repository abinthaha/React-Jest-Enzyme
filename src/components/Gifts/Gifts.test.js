import React from 'react';
import { shallow } from 'enzyme';
import GiftComponent from './Gifts';

describe('Gift', () => {
    const mockRemove = jest.fn();
    const id = 1;
    const props = { gift: { id }, removeGift: mockRemove }
    const gift = shallow(<GiftComponent {...props} />);

    it('Renders correctly', () => {
        expect(gift).toMatchSnapshot();
    })

    it('Initializes a person and present inside `state`', () => {
        expect(gift.state()).toEqual({person: '', present: ''})
    })

    describe('On typing in person input', () => {
        const person= 'Uncle';
        beforeEach(() => {
            gift.find('.input-person').simulate('change', { target: { value: person }});
        })

        it('Updates the person', () => {
            expect(gift.state().person).toEqual(person);
        })
    })

    describe('On typing in present input', () => {
        const present= 'Dinner';
        beforeEach(() => {
            gift.find('.input-present').simulate('change', { target: { value: present }});
        })

        it('Updates the present', () => {
            expect(gift.state().present).toEqual(present);
        })
    })

    describe('On clicking the `remove gift` button', () => {
        beforeEach(() => {
            gift.find('.btn-remove').simulate('click');
        })

        it ('calls the `removeGift` callback', () => {
            expect(mockRemove).toHaveBeenCalledWith(id);
        })
    })
})