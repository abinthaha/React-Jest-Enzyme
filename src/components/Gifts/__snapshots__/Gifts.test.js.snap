// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift Renders correctly 1`] = `
<div
  className="each-gift"
>
  <ForwardRef(Bootstrap(Form))>
    <ForwardRef(Bootstrap(FormGroup))>
      <ForwardRef(ContextTransform)>
        Person
      </ForwardRef(ContextTransform)>
      <ForwardRef(ContextTransform)
        className="input-person"
        onChange={[Function]}
      />
    </ForwardRef(Bootstrap(FormGroup))>
    <ForwardRef(Bootstrap(FormGroup))>
      <ForwardRef(ContextTransform)>
        Present
      </ForwardRef(ContextTransform)>
      <ForwardRef(ContextTransform)
        className="input-present"
        onChange={[Function]}
      />
    </ForwardRef(Bootstrap(FormGroup))>
  </ForwardRef(Bootstrap(Form))>
  <ForwardRef(Bootstrap(Button))
    className="btn-remove"
    onClick={[Function]}
  >
    Remove Gift
  </ForwardRef(Bootstrap(Button))>
</div>
`;
